### PR TITLE
publish jvm artifacts in mpp maven module

### PR DESCRIPTION
**Pull request by i-walker**
I've adopted the same pattern from [kotlin.coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/3574c2feca23c3e8a1ad00b5bf92e2bf04d95060/gradle/publish.gradle#L72)
Can we test this on a library first?
Does the formatting from spotlessApply seem a bit off here?

### CODE REVIEW

**Comment by JavierSegoviaCordoba** [comment]
Gradle has a lot of lambdas and the we usually see them expanded, meanwhile spotless put them in a single line if it is possible so it is normal to see Gradle things weird with spotless.

**Comment by JavierSegoviaCordoba** [comment]
For trying it, I recommend you to publish it to maven local and try it in another project.

**Comment by i-walker** [comment]
thanks I will do that :D

**Review by nomisRev — CHANGES_REQUESTED** [review]
@i-walker where did you get this code from?
I think the setup that was missing is wiring this to the actual publishing configuration defined here: https://github.com/arrow-kt/arrow-gradle-config/blob/main/arrow-gradle-config-publish/src/main/kotlin/internal/ConfigurePublish.kt

I'm not sure what these changes do in addition to what was already there? The file you changed is this file: https://github.com/Kotlin/kotlinx.coroutines/blob/3574c2feca23c3e8a1ad00b5bf92e2bf04d95060/gradle/publish-mpp-root-module-in-platform.gradle

I think these changes are incorrect.

File: arrow-gradle-config-publish/src/main/kotlin/internal/ConfigurePublish.kt [hunk 1]
```diff
@@ -86,6 +87,20 @@ fun Project.configurePublishing(
...
 
+          all {
+            when (name) {
+              "kotlinMultiplatform" -> {
+                // the root mpp module ID has no suffix, but for compatibility with the consumers who
+                // can't read Gradle module metadata, we publish the JVM artifacts in it
+                artifactId = project.name
+                publishPlatformArtifactsInRootModule(publications.getByName<MavenPublication>("jvm"))
+              }
+              "metadata", "jvm", "js" -> {
```
Comment (nomisRev) [inline arrow-gradle-config-publish/src/main/kotlin/internal/ConfigurePublish.kt:101]:
```suggestion
else -> {
```

Maybe an `else` case here is better.
In KotlinX Coroutines all "native" platforms are renamed to `native`, so if you're on `windows` then they only enable the `mwing` target and rename it to `native`.

**Review by nomisRev — APPROVED** [review]
Looks great @i-walker. I think we can merge and try this!

**Pull request merged**